File: Java/problem1071.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        int sLen1 = str1.length(), sLen2 = str2.length();
        if (sLen1 < sLen2) {
            return gcdOfStrings(str2, str1);
        }
        int n = sLen1 / sLen2;
        for (int i = 0; i < n; ++i) {
            if (!str2.equals(str1.substring(i * sLen2, (i + 1) * sLen2))) {
                return "";
            }
        }
        if (sLen1 % sLen2 == 0) {
            return str2;
        }
        return gcdOfStrings(str2, str1.substring(n * sLen2));
    }
}

File: python/problem1071.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        sLen1, sLen2 = len(str1), len(str2)
        if sLen1 < sLen2:
            return self.gcdOfStrings(str2, str1)
        n = sLen1 // sLen2
        for i in range(n):
            if not str2 == str1[i * sLen2 : (i + 1) * sLen2]:   #取模操作前需判断是否能取模
                return ''
        if sLen1 % sLen2 == 0:
            return str2
        return self.gcdOfStrings(str2, str1[n * sLen2 :])       #第二个参数即为取模之后的字符串
